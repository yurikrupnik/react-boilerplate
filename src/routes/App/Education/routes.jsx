import Diamonds from './Diamonds/index';
import Engagement from './Engagement/index';
import Sidebar from './Sidebar/index';
import Main from './Main/index';
import Router from '../../services/Router';

const routes = [
    {
        path: '/education',
        component: Main,
        exact: true
    },
    {
        path: '/education/diamonds',
        main: Diamonds,
        sidebar: Sidebar,
        label: 'loose diamonds'
    },
    {
        path: '/education/engagement-rings',
        main: Engagement,
        sidebar: Sidebar,
        label: 'engagement rings'
    }
];

export default Router.create(routes);
