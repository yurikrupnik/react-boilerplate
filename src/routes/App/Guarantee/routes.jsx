import uniqid from 'uniqid';
import React from 'react';
import { Redirect } from 'react-router-dom';
import CustomerService from './CustomerService/index';
import SideBar from './SideBar/index';
import GuaranteeContextComp from './GuaranteeContext/index';

const routes = [
    {
        key: uniqid(),
        path: '/guarantee',
        component: () => (<Redirect to="/guarantee/customer-service" />),
        exact: true
    },
    {
        key: uniqid(),
        path: '/guarantee/customer-service',
        component: CustomerService,
        sidebar: SideBar
    },
    {
        key: uniqid(),
        path: '/guarantee/:id',
        component: GuaranteeContextComp,
        sidebar: SideBar
    }
];

export default routes;
