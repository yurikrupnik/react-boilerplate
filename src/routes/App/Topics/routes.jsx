import uniqid from 'uniqid';
import React from 'react';
import PropTypes from 'prop-types';

const Topic = ({ match }) => (
    <div>
        <h3>
            {match.params.topicId}
        </h3>
    </div>
);

Topic.propTypes = {
    match: PropTypes.shape({}).isRequired
};

const routes = [
    {
        key: uniqid(),
        path: '/topics',
        component: () => (
            <h3>
                Please select a topic.
            </h3>
        ),
        exact: true
    },
    {
        key: uniqid(),
        path: '/topics/:topicId',
        component: Topic,
        exact: true
    }
];

export default routes;
