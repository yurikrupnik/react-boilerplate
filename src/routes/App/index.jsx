import Loadable from '../../components/Loadable';

const App = Loadable({
    loader: () => import('./app'),
});

export default App;
