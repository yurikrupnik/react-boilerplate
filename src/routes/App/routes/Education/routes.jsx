import uniqid from 'uniqid';
import Diamonds from './Diamonds';
import Engagement from './Engagement';
import Sidebar from './Sidebar';
import Main from './Main';

const routes = [
    {
        path: '/education',
        component: Main,
        exact: true,
        key: uniqid()
    },
    {
        path: '/education/diamonds',
        main: Diamonds,
        sidebar: Sidebar,
        label: 'loose diamonds',
        key: uniqid()
    },
    {
        path: '/education/engagement-rings',
        main: Engagement,
        sidebar: Sidebar,
        label: 'engagement rings',
        key: uniqid()
    }
];


function getRoutesByType(isMobile) {
    return routes.reduce((acc, next) => {
        if (next.component) {
            return acc.concat(Object.assign({}, next, {
                component: Array.isArray(next.component) ? isMobile ? next.component[0] : next.component[1] : next.component // eslint-disable-line
            }));
        }
        if (next.main) {
            return acc.concat(Object.assign({}, next, {
                main: Array.isArray(next.main) ? isMobile ? next.main[0] : next.main[1] : next.main // eslint-disable-line
            }));
        }
        if (next.sidebar) {
            return acc.concat(Object.assign({}, next, {
                sidebar: Array.isArray(next.sidebar) ? isMobile ? next.sidebar[0] : next.sidebar[1] : next.sidebar // eslint-disable-line
            }));
        }
        return acc.concat(next);
    }, []);
}

function getLinks() {
    return routes.reduce((acc, route) => {
        if (!route.label) {
            return acc;
        }
        return acc.concat({
            label: route.label,
            key: route.key,
            to: route.path
        });
    }, []);
}

export default { getRoutesByType, getLinks };
