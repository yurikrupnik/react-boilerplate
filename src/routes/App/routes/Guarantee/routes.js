import uniqid from 'uniqid';
import React from 'react';
import { Redirect } from 'react-router-dom';
import CustomerService from './CustomerService/index';
import LifetimeWarranty from './LifetimeWarranty/index';

// import Shipping from './Shipping';
// import Engraving from './Engraving';
// import Returns from './Returns';
// import Update from './Update';
import SlideBar from './SlideBar';
import GuaranteeContextComp from './GuaranteeContextComp';

const routes = [
    {
        path: '/guarantee',
        component: () => (<Redirect to="/guarantee/customer-service" />),
        exact: true,
        key: uniqid()
    },
    {
        path: '/guarantee/customer-service',
        component: CustomerService,
        sidebar: SlideBar,
        key: uniqid(),
        label: '24/7 Customer Service',
        headerLabel: '24/7 Customer Service',
    },
    {
        path: '/guarantee/:id',
        component: GuaranteeContextComp,
        sidebar: SlideBar,
        key: uniqid(),
        label: 'Lifetime Warranty',
        headerLabel: 'Lifetime Warranty',
    }
    // {
    //     path: '/guarantee/lifetime-warranty',
    //     component: LifetimeWarranty,
    //     sidebar: SlideBar,
    //     key: uniqid(),
    //     label: 'Lifetime Warranty',
    //     headerLabel: 'Lifetime Warranty'
    // },
    // {
    //     path: '/guarantee/free-shipping',
    //     component: Shipping,
    //     sidebar: SlideBar,
    //     key: uniqid(),
    //     label: 'Free Shipping Worldwide',
    //     headerLabel: 'FREE INTERNATIONAL SHIPPING'
    // },
    // {
    //     path: '/guarantee/hassle-free-returns',
    //     component: Returns,
    //     sidebar: SlideBar,
    //     key: uniqid(),
    //     label: 'Hassle-Free Returns',
    //     headerLabel: '100% MONEY BACK GUARANTEE',
    //     extraLabel: ' with 100% Money Back Guarantee'
    // },
    // {
    //     path: '/guarantee/free-engraving',
    //     component: Engraving,
    //     sidebar: SlideBar,
    //     key: uniqid(),
    //     label: 'Free Engraving'
    // },
    // {
    //     path: '/guarantee/lifetime-upgrade',
    //     component: Update,
    //     sidebar: SlideBar,
    //     key: uniqid(),
    //     label: 'Lifetime Upgrade'
    // }
];

function getRoutesByType(isMobile) {
    return routes.reduce((acc, next) => {
        if (next.component) {
            return acc.concat(Object.assign({}, next, {
                component: Array.isArray(next.component) ? isMobile ? next.component[0] : next.component[1] : next.component
            }));
        }
        if (next.main) {
            return acc.concat(Object.assign({}, next, {
                main: Array.isArray(next.main) ? isMobile ? next.main[0] : next.main[1] : next.main
            }));
        }
        if (next.sidebar) {
            return acc.concat(Object.assign({}, next, {
                sidebar: Array.isArray(next.sidebar) ? isMobile ? next.sidebar[0] : next.sidebar[1] : next.sidebar
            }));
        }
        return acc.concat(next);
    }, []);
}

function getLinks() {
    return routes.reduce((acc, route) => {
        if (!route.label) {
            return acc;
        }
        return acc.concat({
            label: route.label,
            key: route.key,
            to: route.path
        });
    }, []);
}

export default { getRoutesByType, getLinks };
