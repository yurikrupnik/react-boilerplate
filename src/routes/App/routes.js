import uniqid from 'uniqid';
import News from './routes/Newsletter';
import Dashboard from './routes/Dashboard';
import Header from './routes/Header';
import Guarantee from './routes/Guarantee';
import Topics from '../../components/Topics';
import Education from './routes/Education';

const routes = [
    {
        path: '/',
        component: Header,
        key: uniqid(),
        fetch: () => {
            return Promise.resolve([
                {
                    user: 'shit',
                    lol: true
                }
            ]);
        }
    },
    {
        path: '/',
        component: Dashboard,
        exact: true,
        key: uniqid()
    },
    {
        component: Guarantee,
        path: '/guarantee',
        label: 'guarantee',
        key: uniqid()
    },
    {
        component: Education,
        path: '/education',
        label: 'education',
        key: uniqid()
    },
    {
        component: Topics,
        path: '/topics',
        label: 'topics',
        key: uniqid()
    },
    {
        path: '/',
        component: News,
        key: uniqid()
    }
];

function getRoutesByType(isMobile) {
    return routes.reduce((acc, next) => {
        if (next.component) {
            return acc.concat(Object.assign({}, next, {
                component: Array.isArray(next.component) ? isMobile ? next.component[0] : next.component[1] : next.component // eslint-disable-line
            }));
        }
        if (next.main) {
            return acc.concat(Object.assign({}, next, {
                main: Array.isArray(next.main) ? isMobile ? next.main[0] : next.main[1] : next.main // eslint-disable-line
            }));
        }
        if (next.sidebar) {
            return acc.concat(Object.assign({}, next, {
                sidebar: Array.isArray(next.sidebar) ? isMobile ? next.sidebar[0] : next.sidebar[1] : next.sidebar // eslint-disable-line
            }));
        }
        return acc.concat(next);
    }, []);
}

function getLinks() {
    return routes.reduce((acc, route) => {
        if (!route.label) {
            return acc;
        }
        return acc.concat({
            label: route.label,
            key: route.key,
            to: route.path
        });
    }, []);
    // console.log('links', links);

    // return [
    //     {
    //         to: '/asd',
    //         key: 'asd',
    //         label: 'omg'
    //     }
    // ];
}

export default { routes, getRoutesByType, getLinks };
