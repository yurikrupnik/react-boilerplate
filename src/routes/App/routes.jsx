import uniqid from 'uniqid';
import Dashboard from './Dashboard';
import Guarantee from './Guarantee';
import Topics from './Topics';
import Education from './Education';
import Header from './Header';
import Newsletter from './Newsletter';
import api from '../../api/currency/api';
import users from '../../api/users/api';

const keyByElement = (ele) => {
    if (Array.isArray(ele)) {
        return ele[0].name;
    }
    return ele.name;
};

const routes = [
    {
        key: 'header',
        path: '/',
        component: Header
    },
    {
        key: keyByElement(Dashboard),
        path: '/',
        component: Dashboard,
        exact: true,
        // fetchInitialData: (url) => {
        //     console.log('url', url); // eslint-disable-line no-console
        //     return Promise.all([
        //         api.fetch(), users.fetch()
        //     ]);
        // },
        // providers: [api.provider, users.provider]
    },
    {
        key: keyByElement(Guarantee),
        component: Guarantee,
        path: '/guarantee',
        label: 'guarantee',
    },
    {
        key: keyByElement(Education),
        component: Education,
        path: '/education',
        label: 'education',
    },
    {
        key: keyByElement(Topics),
        component: Topics,
        path: '/topics',
        label: 'topics',
    },
    {
        key: 'newsletter',
        path: '/',
        component: Newsletter
    }
];

export default routes;
