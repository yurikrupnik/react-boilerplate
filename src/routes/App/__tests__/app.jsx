import React from 'react';
import App from '../app';
import routes from '../routes';

jest.mock('../routes'); // eslint-disable-line no-undef
const {
    it,
    expect,
    shallow,
    navigator
} = global;

it('renders <App /> component', () => {
    const wrapper = shallow(<App userAgent={navigator.userAgent} routes={routes} />);
    expect(wrapper).toMatchSnapshot();
});
