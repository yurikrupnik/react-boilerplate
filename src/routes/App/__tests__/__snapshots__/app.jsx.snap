// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App /> component 1`] = `
<Providers
  providers={
    Array [
      [Function],
      [Function],
      [Function],
      [Function],
    ]
  }
>
  <DeviceConsumer
    render={[Function]}
  />
</Providers>
`;
